src/ML-POD/compute_podd_atom.cpp

```diff
@@ -58,6 +58,8 @@ ComputePODDAtom::ComputePODDAtom(LAMMPS *lmp, int narg, char **arg) :
   pod = nullptr;
   elements = nullptr;
 
+  if (((((MAXBIGINT*3.0)*atom->natoms)*podptr->nClusters)*podptr->Mdesc) > (MAXSMALLINT*1.0))
```

Comment:
@exapde the statement below in line 63 was flagged by static code analysis as potentially overflowing a 32-bit integer. So I added a check, but please be aware that what you are doing here can potentially "strangle" machines, since your memory consumption will grow quadratically or possibly even worse with the number of atoms.

You want to store per-atom data (so that is going to consume `sizeof(double)*Atom::nlocal*size_peratom_cols`). Thanks to domain decomposition the amount of per-processor storage can be reduced by using more processors. However, you also multiply with `Atom::natoms` which is the global number of atoms and then have multiple datums per-atom, some of which look like they could also grow as the total number of atoms grows. This can very, very quickly lead to huge memory consumption for all but the smallest systems with a small number of atoms.

Reply:
@akohlmey Thanks a lot for doing this. 

The purpose of these compute classes is to train the POD potential on DFT datasets, which are typically a few hundred atoms at most. They are never used to run MD simulations. 

I use Atom::natoms to store per-atom data mainly because LAMMPS does not provide MPI assembly for the compute classes. I have to tally the per-atom descriptors into the global descriptors, similar to how forces are tallied. While I can do that with the pair-pod class, I could not do with the compute pod classes.

Reply:
> While I can do that with the pair-pod class, I do not know how to do the same with the compute pod classes.

This is something where you should ask the LAMMPS developers. Do you have access to the LAMMPS developer slack? That is the place to ask.

You can do a reverse communication with computes just as well as it is done (automatically) after the forces are computed. You just needed to define a pack_reverse_comm() and unpack_reverse_comm() function and flag how many items you need to collect from ghosts to the principal atoms.
There are over 15 such computes in the LAMMPS distribution currently.
You should also check out the recent talk by Steve on the "Master Class" workshop: https://www.youtube.com/@templelammps/streams, i.e.  https://www.youtube.com/live/fhNhZ6ilTOU?feature=shared